### ISSUE TITLE
Unable to catch assureDioError using try-catch

### ISSUE THREAD

**Issue by DenzelCode** [flutterchina/dio#1228]
### New Issue Checklist

- [] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |      |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps
It happens whenever I return a statusCode which refers to an error on my api, I'm not being able to catch the assureDioError which always keep being pointed at on my debug console.

![image](https://user-images.githubusercontent.com/27902328/126220852-23b7b2e8-ef55-44fc-99bd-e9aa88a08159.png)

![image](https://user-images.githubusercontent.com/27902328/126220705-453adebe-7dc1-45c2-b4a3-6d8e5a3295e3.png)

![image](https://user-images.githubusercontent.com/27902328/126220586-c591fb05-1815-4267-b2ec-af7e2f83abfa.png)

**Comment by dhsont** [comment]
Same happening with me.

**Comment by sysquo-aditya** [comment]
+1 
execution never reaches the catch block.

**Comment by yagoagrosmart** [comment]
+1

Same here.

**Comment by zhangKunUserGit** [comment]
+1

**Comment by zhangKunUserGit** [comment]
Hope to solve it quickly, the `onError` of `interceptors`  will not be triggered

**Comment by JobMoll** [comment]
+1

**Comment by huanghongxing-jim** [comment]
+1

**Comment by TheAlphamerc** [comment]
I was facing the same issue but I found out why I wasn't able catch `DioError` in try-catch block while calling the APi.
For those who are facing the issue, I am posting how It is solved.

### Dio client configuration
<img width="434" alt="Screenshot 2021-07-28 at 10 42 21 AM" src="https://user-images.githubusercontent.com/37103237/127267636-1dfbb6fa-f782-4661-a22a-46b9bb93c74d.png">


### Method from where I am calling API.
<img width="576" alt="Screenshot 2021-07-28 at 10 49 52 AM" src="https://user-images.githubusercontent.com/37103237/127267950-3a55b06e-11a2-4391-a2fc-3a476cd6c57a.png">


### AppInterceptors class
I forgot to call `super.onError(dioError, handler);` from `onError` method. After addding this code I was able to catch error in `post` method shared in above screenshot.

<img width="833" alt="Screenshot 2021-07-28 at 10 41 31 AM" src="https://user-images.githubusercontent.com/37103237/127267708-5fbce3f3-3f5b-4208-b0aa-f6fa091ac494.png">

**Comment by yagoagrosmart** [comment]
> I forgot to call `super.onError(dioError, handler);` from `onError` method. After addding this code I was able to catch error in `post` method shared in above screenshot.

This work for me, thanks @TheAlphamerc

**Comment by novas1r1** [comment]
@TheAlphamerc you are my life-saver <3! Works perfectly

**Comment by sysquo-aditya** [comment]
I have just downgraded to one of the beta versions. No errors so far.

Just seems unnecessary to have to write a new interceptor just to catch errors when previous versions have been doing that already.

**Comment by Williano** [comment]
@sysquo-aditya which beta version did you use?
